junit-test-result-parser/src/main/java/org/arquillian/smart/testing/parser/junit/JUnitTestResultParser.java

```diff
@@ -0,0 +1,95 @@
+
+        try {
+            final XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+            XMLEventReader eventReader = inputFactory.createXMLEventReader(surefireInputStream);
+
+            TestResult currentTestResult = null;
+
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+
+                if (event.isStartElement()) {
+                    final StartElement startElement = event.asStartElement();
+                    if ("testcase".equalsIgnoreCase(startElement.getName().getLocalPart())) {
+                        // Read attributes
+                        String name = null, classname = null, duration = null;
+
+                        final Iterator<Attribute> attributes = startElement.getAttributes();
+
+                        while (attributes.hasNext()) {
+                            final Attribute attribute = attributes.next();
+                            if ("classname".equalsIgnoreCase(attribute.getName().toString())) {
+                                classname = attribute.getValue();
+                            }
+
+                            if ("name".equalsIgnoreCase(attribute.getName().toString())) {
+                                name = attribute.getValue();
+                            }
+
+                            if ("time".equalsIgnoreCase(attribute.getName().toString())) {
+                                duration = attribute.getValue();
+                            }
+                        }
+
+                        final float durationInSeconds = Float.parseFloat(duration);
+                        currentTestResult = new TestResult(classname, name, durationInSeconds);
+                    }
+
+                    if ("failure".equalsIgnoreCase(startElement.getName().getLocalPart())) {
+                        currentTestResult.setResult(TestResult.Result.FAILURE);
+                    }
+
+                    if ("error".equalsIgnoreCase(startElement.getName().getLocalPart())) {
+                        currentTestResult.setResult(TestResult.Result.ERROR);
+                    }
+
+                    if ("skipped".equalsIgnoreCase(startElement.getName().getLocalPart())) {
+                        currentTestResult.setResult(TestResult.Result.SKIPPED);
+                    }
+
+                    if ("rerunFailure".equalsIgnoreCase(startElement.getName().getLocalPart())) {
+                        currentTestResult.setResult(TestResult.Result.RE_RUN_FAILURE);
+                    }
+                }
+
+                if (event.isEndElement()) {
+                    final EndElement endElementElement = event.asEndElement();
+                    if ("testcase".equalsIgnoreCase(endElementElement.getName().getLocalPart())) {
+                        testResults.add(currentTestResult);
+                    }
+                }
+            }
+        } catch (XMLStreamException e) {
+
```

Comment:
Shouldn't we do something about this exception? It means we have faulty result and we have no clue at the runtime we actually do have a problem.